homeassistant/components/statistics/sensor.py

```diff
@@ -281,22 +281,22 @@ def _next_to_purge_timestamp(self):
 
         # If max_age is set, ensure to update again after the defined interval.
         next_to_purge_timestamp = self._next_to_purge_timestamp()
-        if next_to_purge_timestamp:
+        if (
+            next_to_purge_timestamp is not None
+            and (next_to_purge_timestamp - self._next_purge).total_seconds() > 1
+        ):
+            self._next_purge = next_to_purge_timestamp
             _LOGGER.debug(
-                "%s: scheduling update at %s", self.entity_id, next_to_purge_timestamp
+                "%s: scheduling update at %s", self.entity_id, self._next_purge
             )
-            if self._update_listener:
-                self._update_listener()
-                self._update_listener = None
 
             @callback
             def _scheduled_update(now):
                 """Timer callback for sensor update."""
                 _LOGGER.debug("%s: executing scheduled update", self.entity_id)
                 self.async_schedule_update_ha_state(True)
-                self._update_listener = None
 
-            self._update_listener = async_track_point_in_utc_time(
+            async_track_point_in_utc_time(
```

Comment:
By removing the listener (and its associated code to remove that listener), there is a chance that multiple listeners are being created for the same sensor. It's not too bad, because there should always be a listener for the very next timestamp, however (if many entities are observed) we could end up with many superfluous listeners at the same time that just use up compute time without changing the `statistics` sensor state.

Reply:
yeah, I saw this problem, I'll revert it, thanks!